src/ASM/ASMs2Dmx.C

```diff
@@ -1162,6 +1126,12 @@ void ASMs2Dmx::swapProjectionBasis ()
 }
 
 
+int ASMs2Dmx::getFirstItgElmNode () const
+{
+  return std::accumulate(elem_size.begin(), elem_size.begin() + geoBasis-1, 0);
+}
+
+
```

Comment:
This implementation (as well as for getLastItgElmNode below) could also have been places in the common base ASMmxBase, since they only use members of that class, and the 3D version is identical. But you still need to forward to that method from here though (consider if worthwhile).

Reply:
yeah i know but i decided it was just an extra jump with little gain.